x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/CsvTestsDataLoader.java

```diff
@@ -236,15 +242,45 @@ public static void loadDataSetIntoEs(RestClient client, Logger logger) throws IO
     }
 
     private static void loadDataSetIntoEs(RestClient client, Logger logger, IndexCreator indexCreator) throws IOException {
+        if (hasCapability(client, EsqlCapabilities.Cap.SEMANTIC_TEXT_TYPE)) {
+            createInferenceEndpoint(client);
+        }
+
+        Set<String> loadedDatasets = new HashSet<>();
         for (var dataset : CSV_DATASET_MAP.values()) {
+            if (dataset.requiredCapability != null && hasCapability(client, dataset.requiredCapability) == false) {
+                continue;
+            }
             load(client, dataset, logger, indexCreator);
+            loadedDatasets.add(dataset.indexName);
         }
-        forceMerge(client, CSV_DATASET_MAP.keySet(), logger);
+        forceMerge(client, loadedDatasets, logger);
         for (var policy : ENRICH_POLICIES) {
             loadEnrichPolicy(client, policy.policyName, policy.policyFileName, logger);
         }
     }
 
+    private static void createInferenceEndpoint(RestClient client) throws IOException {
+        Request request = new Request("PUT", "_inference/sparse_embedding/test_sparse_inference");
+        request.setJsonEntity("""
+                  {
+                   "service": "test_service",
```

Comment:
this is the inference service from the `inference-service-test` test plugin

Reply:
This one's probably worth javadoc to explain that it's for the semantic text fields.

Reply:
It kinda surprises me that this is the first capability conditional in the test infra.  How expensive is this service endpoint? Conditionally creating it is of course fine, but the test infra would be simpler if it was unconditionally registered.

Reply:
The test endpoint is quite lightweight. We could always register it.

Reply:
javadoc was added